sw/device/lib/base/memory.c

```diff
@@ -12,26 +14,109 @@
 #define OT_PREFIX_IF_NOT_RV32(name) ot_##name
 #endif
 
+static size_t compute_num_leading_bytes(uintptr_t left, uintptr_t right,
+                                        size_t len) {
```

Comment:
nit: how does `void *` sound instead of `uintptr_t` for `left` and `right`? We could cast only in a few places and call sites would be cleaner.